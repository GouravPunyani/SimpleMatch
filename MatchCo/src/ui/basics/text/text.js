// @flow

import React from 'react'
import { Text as ReactNativeText, StyleSheet } from 'react-native'
import { fontFamily, color, palette } from '../../theme'
import { omit } from 'ramda'
import { translator } from '../../../services/translate/translator'

const withoutKnownProps = omit(['text', 'children', 'style', 'small', 'title', 'tx'])

export type Props = {
  text?: string,
  children?: React.Children,
  style?: any,
  small?: Boolean,
  title?: Boolean,
  fancy?: Boolean,
  brand?: Boolean,
  lighter? : Boolean,
  white?: Boolean,
  tx?: string,
  translate: (key: string) => string,
  center?: Boolean
};

export const Text = translator((props: Props) => {
  const style = [styles.text]
  if (props.small) {
    style.push(styles.small)
  }
  if (props.title) {
    style.push(styles.title)
  }
  if (props.style) {
    style.push(props.style)
  }
  if (props.fancy) {
    style.push(styles.fancy)
  }
  if (props.brand) {
    style.push(styles.brand)
  }
  if (props.lighter) {
    style.push(styles.lighter)
  }
  if (props.white) {
    style.push(styles.white)
  }
  if (props.center) {
    style.push(styles.center)
  }
  const textToUse = props.tx ? props.translate(props.tx) : props.text
  return (
    <ReactNativeText style={style} {...withoutKnownProps(props)}>
      {props.children ? props.children : textToUse}
    </ReactNativeText>
  )
})

const styles = StyleSheet.create({
  text: {
    color: color.text,
    fontSize: 14,
    fontFamily: fontFamily.nexa
  },
  small: { fontSize: 12 },
  title: {
    fontSize: 22,
    fontFamily: fontFamily.nexaBold
  },
  fancy: { fontFamily: fontFamily.sangBlueLightItalic },
  brand: { color: color.brand },
  lighter: { color: color.textLighter },
  white: { color: palette.white },
  center: { textAlign: 'center' }
})
