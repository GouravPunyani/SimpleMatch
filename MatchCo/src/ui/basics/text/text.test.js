// @flow

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { Text as Comp } from './text'

test('renders correctly', () => {
  const tree = renderer.create(<Comp text='hi' />)
  expect(tree).toMatchSnapshot()
})
