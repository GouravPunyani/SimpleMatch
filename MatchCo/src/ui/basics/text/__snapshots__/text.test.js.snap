// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#333333",
        "fontFamily": "Nexa",
        "fontSize": 14,
      },
    ]
  }
>
  hi
</Text>
`;
